src/vs/platform/terminal/common/capabilities/commandDetectionCapability.ts

```diff
@@ -461,6 +461,12 @@ export class CommandDetectionCapability extends Disposable implements ICommandDe
 			return prompt;
 		}
 
+		// Custom prompts like starship end in the common \u276f character
+		const customMatch = lineText.match(/.*\u276f(?=[^\u276f]*\s?$)/g);
+		if (customMatch) {
+			return customMatch[0];
+		}
```

Comment:
This isn't quite what I meant, you need to do the dance with ends with like the powershell part in order to get the real most likely start position:

```ts
			if (lineText === prompt && prompt.endsWith('>')) {
				// Conpty may not 'render' the space at the end of the prompt
				prompt += ' ';
			}
```